File: querydsl_practice/src/main/java/io/github/leeseojune53/querydsl_practice/controller/UserController.java
package io.github.leeseojune53.querydsl_practice.controller;

import io.github.leeseojune53.querydsl_practice.dto.RegisterRequest;
import io.github.leeseojune53.querydsl_practice.service.UserService;
import lombok.RequiredArgsConstructor;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class UserController {

	private final UserService userService;

	@PostMapping("/register")
	public void register(@RequestBody RegisterRequest request) {
		userService.register(request);
	}

	@PostMapping("/signup")
	public String signup(@RequestBody RegisterRequest request) {
		return userService.signUp(request);
	}

}

File: BackEnd_Learn/kotlin/kopring/src/main/kotlin/io/github/leeseojune53/kopring/domain/user/presentation/UserController.kt
package io.github.leeseojune53.kopring.domain.user.presentation

import io.github.leeseojune53.kopring.domain.user.presentation.dto.request.UserRequest
import io.github.leeseojune53.kopring.domain.user.presentation.dto.response.TokenResponse
import io.github.leeseojune53.kopring.domain.user.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
class UserController(
    private val userService: UserService,
) {

    @PostMapping("/register")
    @ResponseStatus(HttpStatus.CREATED)
    fun signUp(@RequestBody request: UserRequest) {
        userService.saveUser(request)
    }

    @PostMapping("/login")
    fun signIn(@RequestBody request: UserRequest): TokenResponse {
        return userService.login(request)
    }

    @PutMapping("/refresh")
    fun tokenRefresh(@RequestHeader("Refresh-Token") token: String): TokenResponse {
        return userService.tokenRefresh(token);
    }

}